compiler/tflchef/core/src/CustomOp/MaxPoolWithArgMax.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ * Copyright 2015 The TensorFlow Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "MaxPoolWithArgMax.h"
+
+#include "flatbuffers/flexbuffers.h"
+
+flatbuffers::Offset<void> MaxPoolWithArgMaxChef::value(flatbuffers::FlatBufferBuilder &fbb) const
+{
+  return flatbuffers::Offset<void>();
+}
+
+flatbuffers::Offset<flatbuffers::Vector<uint8_t>>
+MaxPoolWithArgMaxChef::custom_value(flatbuffers::FlatBufferBuilder &fbb) const
+{
+  auto &operation = (*_operation);
+
+  assert(operation.type() == "MaxPoolWithArgMax");
+
+  /**
+   * REGISTER_OP("MaxPoolWithArgmax")
+    .Attr("ksize: list(int) >= 4")
+    .Attr("strides: list(int) >= 4")
+    .Attr("Targmax: {int32, int64} = DT_INT64")
+    .Attr(GetPaddingAttrString())
+    .Attr("include_batch_in_index: bool = false")
+    .Input("input: T")
+    .Output("output: T")
+    .Output("argmax: Targmax")
+    .Attr("T: realnumbertype")
+    .SetShapeFn([](InferenceContext* c) {
+      TF_RETURN_IF_ERROR(shape_inference::MaxPoolShape(c));
+      c->set_output(1, c->output(0));
+      return Status::OK();
+    });
+   */
+
+  auto flex_buffers = std::make_unique<flexbuffers::Builder>();
+  size_t map_start = flex_buffers->StartMap();
+
+  auto start = flex_buffers->StartVector("ksize");
+  flex_buffers->Add(1);
+  flex_buffers->Add(operation.max_pool_with_argmax_options().filter_width());
+  flex_buffers->Add(operation.max_pool_with_argmax_options().filter_height());
+  flex_buffers->Add(1);
+  flex_buffers->EndVector(start, /*typed=*/true, /*fixed=*/false);
+  start = flex_buffers->StartVector("strides");
+  flex_buffers->Add(1);
+  flex_buffers->Add(operation.max_pool_with_argmax_options().stride_w());
+  flex_buffers->Add(operation.max_pool_with_argmax_options().stride_h());
+  flex_buffers->Add(1);
+  flex_buffers->EndVector(start, /*typed=*/true, /*fixed=*/false);
+  std::string padding = operation.max_pool_with_argmax_options().padding() ? "VALID" : "SAME";
+  flex_buffers->String("padding", padding);
+  auto output_type = operation.max_pool_with_argmax_options().output_type();
+  assert(output_type == tflchef::TensorType::INT32 || output_type == tflchef::TensorType::INT64);
+  int32_t ot = (output_type == tflchef::TensorType::INT32) ? 0 : 1;
+  flex_buffers->Int("Targmax", ot);
```

Comment:
It would be nice if you can put `Targmax` and `include_batch_in_index` in the sequence of above ` REGISTER_OP("MaxPoolWithArgmax")` comments to follow this code block.

Reply:
FYI
I want to remove Targmax attribute to comply with raw custom options format (see [comment](https://github.com/Samsung/ONE/pull/4719#discussion_r512342559))

Reply:
OK

Reply:
Ah, wait. It was not present in 2.0.x TFs, but in 2.3.x it is generated.

Actually TF team added another `T` attribute =/

What version should we comply with?

If there is no such agreement, I'll stick to 2.3, because it looks like additional attributes does not break anything really. (at least in our code)

Reply:
> What version should we comply with? I'll stick to 2.3

Current unofficial version we use is 2.3.0. OK to go :)

Reply:
Note: about value of this attribute: do not save direct number, use `tflite::TensorType_FLOAT32` or `tflite::TensorType_INT64` instead